cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -0,0 +1,123 @@
+use sc_client_api::BlockBackend;
+use sp_api::ProvideRuntimeApi;
+use sp_runtime::{generic::BlockId, traits::Block as BlockT};
+
+use cirrus_node_primitives::ProcessorResult;
+use cirrus_primitives::{AccountId, SecondaryApi};
+use sp_executor::{ExecutionReceipt, OpaqueBundle};
+use subspace_runtime_primitives::Hash as PHash;
+
+use super::{Executor, LOG_TARGET};
+use codec::{Decode, Encode};
+
+impl<Block, BS, RA, Client, TransactionPool> Executor<Block, BS, RA, Client, TransactionPool>
```

Comment:
Honestly I find it really confusing when implementation is moved into a separate file

Reply:
Well, I don't see it as uncommon. I foresee `lib.rs` will soon become bloated so I'd like to move the impl of `bundler` and `processor` to another separate module. Will renaming to `impl_processor.rs` make you slightly better?

Reply:
It is more about having the whole impl for some struct/enum in one file, not naming of the file.

I need to open the file and being able to read it all. If file is like 5000 lines then probably a better abstraction is needed, but moving functions into separate files hardly helps understanding it, it just pretends to be less complicated than it actually is and one can easily not recognize that there is part of the implementation in the other file.

Reply:
>I need to open the file and being able to read it all

With just one click or key mapping(jump to the definition), any modern IDE/editor will do everything for you.

For me, it's better to maintain this way and I have no issue understanding it, but I'm all ears if you have a better abstraction as I believe this file might not reach 5000 lines eventually but can reach 2000+ lines easily.

Reply:
But you need to know what you're looking for. If you just open sidebar with file structure in IDE, it will not show that method because it is not implementing here. Files individually look nice when separated (minus growing generic bounds), but it doesn't help with understanding the logic.

I saw similar situation in some frameworks where every small piece of logic was moved into its own class and file, so it looks nice individually, but you still need to read and put into the head all of it to understand what is going on. The easier it is to discover things that are together the better.

If you want to keep `lib.rs` smaller, leave `start_executor` there and move the rest into `processor.rs`, but I really don't see how moving part of inherent implementation into a separate file makes anything easier to understand.

Reply:
>But you need to know what you're looking for. If you just open sidebar with file structure in IDE, it will not show that method because it is not implementing here. Files individually look nice when separated (minus growing generic bounds), but it doesn't help with understanding the logic.

I don't get this argument, it's untrue for me unless I misunderstand something. Only the internal impls are delegated to another module, the function is still there, so you still see the function(`produce_bundles`) in the file structure `lib.rs`. If you want to know what `produce_bundles` then just click the button and jump to the actual impl function.

Moving to a separate file gives a relatively smaller and specific context assuming you know where you're from and what you want to know. Furthermore, you have to scroll up and down frequently in a pretty large file, which is less efficient for me, it's not an issue if this large file is well organized but easier to achieve that than a smaller file.

Reply:
OK, let's say we add 5 more methods. Will we have 5 more files? Do you really think jumping between 6 files will be more convenient than one (I'm not talking about independent abstracted away things here, just about an impl of the same struct spread out across multiple files).

Scrolling up and down is more efficient that jumping between a bunch of files. A simple example is just doing search in a file and going thorough search results up and down. When results are in different files it makes things more difficult.

Another example is reviewing code on GitHub. You can usually click on sidebar and expand previous or next few lines to see more context, you can also quickly open the whole file and use search there if needed. When impl is in multiple files you need to spend time opening multiple files in different tabs on that particular git revision to understand the whole context, which for is often a necessity for me.

Those would be my 2 biggest reasons to have an implementation in the same file, for me it is very important to be able to fairly accurately tell what is where without absolutely having to use IDE features or else I have to read all of the files in a crate.

Reply:
To be more specific, it is surprisingly rare that I only need one function in isolation, more often I need to know the wider context right away so I can reason about implementation details and do bigger refactoring if things don't quite line up.

Reply:
>Do you really think jumping between 6 files will be more convenient than one

Yes, it's true for me 🤣.  As a vimmer, I don't use the mouse so I normally don't heavily scroll up and down, and indeed actually jump structurally everywhere :P

We're not separating them by the number of methods but by the logic functionalities. Our daily development does not happen on GitHub, so that's not the main concern.

Now I can get the main difference here, I jump back and forth and you scroll up and down 😆

Reply:
Well, everyone scrolls when they review code on GitHub, I usually don't pull the branch into IDE just to review it.

Reply:
There is a Chrome plugin Sourcegraph, might be useful to have some IDE features on GitHub, I don't heavily use it though.